app/src/main/java/com/yoshitoke/weatheringwithyou/mvp/view/MainActivity.kt

```diff
@@ -0,0 +1,158 @@
+
+        location_spinner.setAdapter(cityAdapterObj)
+        location_spinner.onItemSelectedListener = this
+    }
+
+    override fun showWeatherInfo(data: WeatherInfo) {
+        val tempText = data.current.temperature.kelvinToCelsius().toCelsiusString()
+        tv_temperature.setText(tempText)
+
+        tv_dateTime.setText(data.current.dateTime.unixTimestampToString(dateConstant))
+        tv_condition.setText(data.current.weathers[0].description)
+
+        showHourlyForecast(data.hours)
+        initBottomTabs(data)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == REQUEST_CODE_MAPS) {
+
+            if (resultCode == RESULT_OK) {
+                val result = data?.getStringExtra(MapsMarkerActivity.CITY_DATA)
+                val cityObj = Gson().fromJson(result, City::class.java)
+
+                //add local city to database
+                val cityDB = CityDatabaseHandler(this)
+
+                val values = ContentValues()
+                values.put("name", cityObj.name)
+                values.put("latitude", cityObj.latitude)
+                values.put("longitude", cityObj.longitude)
+
+                cityDB.AddCity(values)
+
+                //reload list
+                mPresenter?.loadCityList()
+
+                Toast.makeText(this, "Result: $result", Toast.LENGTH_LONG).show()
+            } else {
+                TODO("RESULT FAILED")
+            }
+        }
+    }
+
+    override fun onDestroy() {
+        mPresenter?.destroy()
+        super.onDestroy()
+    }
+
+    private fun showHourlyForecast(data: List<Hourly>) {
+        mLayoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
+        recyclerView.layoutManager = mLayoutManager
+        mAdapter = HourlyTimeLineAdapter(data)
+        recyclerView.adapter = mAdapter
+    }
+
+    private fun initBottomTabs(data: WeatherInfo) {
+        val viewPagerAdapter = ViewPager2Adapter(this, 2, data)
+        pager.adapter = viewPagerAdapter
+
+        TabLayoutMediator(tab_layout, pager) { tab, position ->
+            when (position) {
+                0 -> tab.text = "Daily forecast"
+                1 -> tab.text = "Additional"
+            }
+        }.attach()
+    }
+
+    private fun initMoreButtonListener() {
+        tabsOpenBtn.setOnClickListener {
+            if(bottom_tabs.visibility == View.GONE) {
+
+                val bottomUp: Animation = AnimationUtils.loadAnimation(this,
+                        R.anim.bottom_up)
+
+                bottom_tabs.startAnimation(bottomUp)
+                bottom_tabs.visibility = View.VISIBLE
+            } else {
+
+                val bottomDown: Animation = AnimationUtils.loadAnimation(this,
+                        R.anim.bottom_down)
+
+                bottom_tabs.startAnimation(bottomDown)
+                bottom_tabs.visibility = View.GONE
+
+            }
```

Comment:
remove some blank lines `136`, `139`, `143`, `146`, `149`
```suggestion
            if(bottom_tabs.visibility == View.GONE) {
                bottom_tabs.apply {
                    startAnimation(AnimationUtils.loadAnimation(this, R.anim.bottom_up))
                    visibility = View.VISIBLE
                }
            } else {
                bottom_tabs.apply {
                    startAnimation(AnimationUtils.loadAnimation(this, R.anim.bottom_down))
                    visibility = View.GONE
                }
            }
```

Reply:
wow